library/core/src/mem/maybe_uninit.rs

```diff
@@ -117,41 +117,36 @@ use crate::slice;
 /// `MaybeUninit<T>` can be used to initialize a large array element-by-element:
 ///
 /// ```
-/// use std::mem::{self, MaybeUninit};
+/// #![feature(maybe_uninit_uninit_array_transpose)]
+/// use std::mem::MaybeUninit;
 ///
 /// let data = {
-///     // Create an uninitialized array of `MaybeUninit`. The `assume_init` is
-///     // safe because the type we are claiming to have initialized here is a
-///     // bunch of `MaybeUninit`s, which do not require initialization.
-///     let mut data: [MaybeUninit<Vec<u32>>; 1000] = unsafe {
-///         MaybeUninit::uninit().assume_init()
-///     };
+///     // Create an uninitialized array of `MaybeUninit`.
+///     let mut data: [MaybeUninit<Vec<u32>>; 1000] = MaybeUninit::uninit().transpose();
```

Comment:
fix: This is a great update, but we can't take it yet.  As `MaybeUninit` is a stable type, it's important that the type-level documentation says how to do things *on stable*.  Once `transpose` is stable then we should absolutely recommend it here, but for now it needs to give people something they can use without nightly.

So please leave this part of the docs as-is, but include an example of using `transpose` like this on the unstable method for nightly users to see.

(This is why `sort`'s documentation didn't mention `sort_unstable` until the latter was stabilized, etc.)

Reply:
Oops, good point. Done.